crates/primitives/transactions/src/from_broadcasted_transactions.rs

```diff
@@ -85,8 +86,239 @@ pub fn try_account_tx_from_broadcasted_tx(
     }
 }
 
+pub fn split_broadcasted_declare_tx_into_declare_tx_and_additional_data(
+    broadcasted_declare_tx: BroadcastedDeclareTransaction,
+    chain_id: Felt252Wrapper,
+) -> Result<(DeclareTransaction, ContractClassData), BroadcastedTransactionConversionError> {
+    fn try_new_declare_transaction(
+        tx: starknet_api::transaction::DeclareTransaction,
+        tx_hash: TransactionHash,
+        class_info: ClassInfo,
+        is_query: bool,
+    ) -> Result<DeclareTransaction, BroadcastedTransactionConversionError> {
+        if is_query {
+            DeclareTransaction::new_for_query(tx, tx_hash, class_info)
+        } else {
+            DeclareTransaction::new(tx, tx_hash, class_info)
+        }
+        .map_err(|_| BroadcastedTransactionConversionError::InvalidTransactionVersion)
+    }
```

Comment:
this code is duplicated below. can move it to an outside function?

Reply:
done